pilot/pkg/model/envoyfilter_test.go

```diff
@@ -129,12 +130,44 @@ func TestConvertEnvoyFilter(t *testing.T) {
 						Proxy: &networking.EnvoyFilter_ProxyMatch{ProxyVersion: `foobar`},
 					},
 				},
+				{ // valid http route patch
+					Patch: &networking.EnvoyFilter_Patch{
+						Operation: networking.EnvoyFilter_Patch_MERGE,
+						Value:     protoconv.MapToStruct(map[string]any{"statPrefix": "bar"}),
```

Comment:
you can use buildPatchStruct here

Reply:
`buildPatchStruct` is also in another package and not exported, so I copied it as an internal function in order to minimize the scope of the PR changes as much as possible.